File: Spotter.Model/Responses/UserSuggestionResponse.cs
namespace Spotter.Model.Responses
{
    public class UserSuggestionResponse
    {
        public int UserId { get; set; }
        public string FullName { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string? CityName { get; set; }
        public int MutualFriendsCount { get; set; }
    }
}

File: UI/spotter_mobile/lib/core/models/user_suggestion_response.dart
class UserSuggestionResponse {
  final int userId;
  final String fullName;
  final String username;
  final String? cityName;
  final int mutualFriendsCount;

  UserSuggestionResponse({
    required this.userId,
    required this.fullName,
    required this.username,
    this.cityName,
    required this.mutualFriendsCount,
  });

  factory UserSuggestionResponse.fromJson(Map<String, dynamic> json) {
    return UserSuggestionResponse(
      userId: json['userId'] as int,
      fullName: json['fullName'] as String,
      username: json['username'] as String,
      cityName: json['cityName'] as String?,
      mutualFriendsCount: json['mutualFriendsCount'] as int,
    );
  }
}
